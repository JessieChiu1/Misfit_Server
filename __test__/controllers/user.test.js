const app = require("../../src/app")
const { MongoMemoryServer } = require("mongodb-memory-server");
const supertest = require("supertest");
const mongoose = require("mongoose")
const User = require("../../src/models/user")

const userPayload = {
    "username": "jessie",
    "password": "password"
}

let userId
let token

describe("user.js", () => {

    beforeAll(async () => {
        mongoServer = await MongoMemoryServer.create();
        await mongoose.connect(mongoServer.getUri());

        // create User
        const response = await supertest(app)
            .post("/api/v1/auth/signup")
            .send(userPayload)

        token = response.body.token

        const foundUser = await User.findOne({
            "username": userPayload.username
        })

        userId = foundUser._id.toString()
    })
    
    afterAll(async () => {
        await mongoose.disconnect();
        await mongoose.connection.close();
    })

    describe("given user does not exist and userId", () => {
        it("should return 404", async () => {
            // act
            const fakeUserId = new mongoose.Types.ObjectId().toString();

            const response = await supertest(app)
            .get(`/api/v1/user/${fakeUserId}`)

            // expectation
            expect(response.status).toBe(404)
            expect(response.body).toMatchSnapshot()
        })
    })

    describe("given user exist and userId", () => {
        it("should return 200 and username if user exists", async () => {
            const response = await supertest(app)
            .get(`/api/v1/user/${userId}`);

            expect(response.status).toBe(200);
            expect(response.body).toHaveProperty("username", userPayload.username);
        });
    })

})
