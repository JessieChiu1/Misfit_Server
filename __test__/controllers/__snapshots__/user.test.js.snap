// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user.js given user does not exist and userId should return 404 1`] = `
{
  "message": "No such user found",
}
`;
